src/lavinmq/queue/stream_queue.cr

```diff
@@ -163,6 +163,7 @@ module LavinMQ
       @msg_store_lock.synchronize do
         stream_queue_msg_store.unmap_segments(except: used_segments)
       end
+      stream_queue_msg_store.drop_overflow
```

Comment:
probably drop segments before trying to unmap them, otherwise double work. And all access to the msg_store has to happen inside its lock. 

```suggestion
        stream_queue_msg_store.drop_overflow
        stream_queue_msg_store.unmap_segments(except: used_segments)
      end
```